database/scripts/schema_converter_migrate.py

```diff
@@ -0,0 +1,197 @@
+                PatientNumber.organization == organization,
+                PatientRecord.sendingfacility == SENDING_FACILITY,
+                PatientRecord.sendingextract == "UKRDC"
+            )
+        )
+
+        results = ukrdc_session.execute(query).scalars().all()
+        if len(results) >1:
+            raise Exception("Ambiguous match in converter")
+        elif len(results) == 1:
+            id_lookups[cache_id] = results[0].pid
+        else:
+            id_lookups[cache_id] = None
+
+    return id_lookups
+
+
+def map_assessments(ukrdc_session, cache_conn, id_map):
+    """Function to map between the assessments table in the cache and ukrdc
+    """
+
+    #grab assessments  
+    # Query to get patient records from cache
+    query = text("""
+        SELECT * 
+        FROM assessment
+        ORDER BY update_date ASC;             
+    """)
+
+
+    results = [result for result in cache_conn.execute(query)]
+    for result in results:
+        pid = id_map.get(result[0])
+        if pid is not None:
+            # construct compound key since we have lost the order they arrived
+            # in we may have to have a slightly different key
+            ass_type = result[6]
+            start_date = result[4]
+            key = f"{pid}:{ass_type}:{start_date}"
+            ukrdc_session.add(
+                Assessment(
+                    id = key, 
+                    pid = pid, 
+
+                    assessmentstart = start_date,
+                    assessmentend = result[5],
+
+                    assessmenttypecode = ass_type,
+                    assessmenttypecodestd = result[7],
+                    assessmenttypecodedesc = result[8],
+
+                    assessmentoutcomecode = result[9],
+                    assessmentoutcomecodestd = result[10],
+                    assessmentoutcomecodedesc = result[11]
+                )
+            )
+
+            try:
+                ukrdc_session.commit() 
+            except Exception as e: 
+                print(e)
+
+    return 
+
+def map_dialysis_prescription(ukrdc_session, cache_conn, id_map):
+    """
+    Function to map and migrate dialysis prescription data from the cache to the UKRDC database.
+
+    Args:
+        ukrdc_session: SQLAlchemy session for the UKRDC database.
+        cache_conn: Connection to the cache database.
+        id_map: Dictionary mapping cache IDs to UKRDC patient PIDs.
+    """
+
+    # Query to get dialysis prescriptions from the cache 
+    query = text("SELECT * FROM dialysisprescription")
+
+    # Execute the query and fetch results
+    cache_results = cache_conn.execute(query)
+
+    for result in cache_results:
+        pid = id_map.get(result[1])  # Map cache ID to UKRDC PID
+
+        if pid is not None:
+            # Construct a unique key for the dialysis prescription
+            key = f"{pid}:{result[2]}:{result[3]}"  # Using `fromtime` and `totime` for uniqueness
```

Comment:
I would check that this key hasn't already been seen, unless there's something preventing it.